src/main/java/au/org/ala/biocache/service/DownloadService.java

```diff
@@ -44,165 +49,295 @@
+     * used by the DownloadService, particularly for testing.
+     * 
+     * @return A new instance of DownloadCreator to be used by {@link DownloadControlThread} instances.
+     */
+    protected DownloadCreator getNewDownloadCreator() {
+        return new DownloadCreatorImpl();
+    }
+    
+    /**
+     * Ensures that all of the download threads are given a chance to shutdown cleanly using thread interrupts when a Spring {@link ContextClosedEvent} occurs.
+     */
+    @Override
+    public void onApplicationEvent(ContextClosedEvent event) {
+        afterInitialisation();
+        if(closed.compareAndSet(false, true)) {
+            try {
+                // Stop more downloads from being added by shutting down additions to the persistent queue
+                persistentQueueDAO.shutdown();
             }
-        }.start();
+            finally {
+                DownloadControlThread nextToCloseRunnable = null;
+                // Call a non-blocking shutdown command on all of the download control threads
+                while((nextToCloseRunnable = runningDownloadControlRunnables.poll()) != null) {
+                    nextToCloseRunnable.shutdown();
+                }
+                
+                Thread nextToCloseThread = null;
+                List<Thread> toJoinThreads = new ArrayList<>();
+                while((nextToCloseThread = runningDownloadControllers.poll()) != null) {
+                    if(nextToCloseThread.isAlive()) {
+                        // Interrupt any threads that are still alive after the non-blocking shutdown command
+                        nextToCloseThread.interrupt();
+                        toJoinThreads.add(nextToCloseThread);
+                    }
+                }
+                
+                if(!toJoinThreads.isEmpty()) {
+                    // Give remaining download control threads a few seconds to cleanup before returning from this method
+                    try {
+                        Thread.sleep(5000);
+                    }
+                    catch(InterruptedException e) {
+                        Thread.currentThread().interrupt();
+                    }
+                }
+            }
+        }
     }
-
+    
     /**
      * Registers a new active download
+     * 
      * @param requestParams
      * @param ip
      * @param type
      * @return
      */
-    public DownloadDetailsDTO registerDownload(DownloadRequestParams requestParams, String ip, DownloadDetailsDTO.DownloadType type){
-        DownloadDetailsDTO dd = new DownloadDetailsDTO(requestParams.toString(), ip, type);
+    public DownloadDetailsDTO registerDownload(DownloadRequestParams requestParams, String ip,
+            DownloadDetailsDTO.DownloadType type) {
+        afterInitialisation();
+        DownloadDetailsDTO dd = new DownloadDetailsDTO(requestParams, ip, type);
+        persistentQueueDAO.addDownloadToQueue(dd);
```

Comment:
Is addDownloadToQueue required here? 

addDownloadToQueue adds requests to offlineDownloadList. This duplicates all online requests as offline making them run twice.